### DISCUSSION TITLE
Expand quantity detection to large units without a "Net" symbol

### DISCUSSION THREAD

**Discussion by CloCkWeRX** [openfoodfacts/robotoff#1233] [Ideas]
Right now, the model seems to heavily rely on the presence of a NET indicator to determine which is most the likely weight.

We should consider looking for units like "1 L" or "1 KG", which are too large to be commonly found in a nutrient table; where no other quantities have been detected.

Examples:
https://world.openfoodfacts.org/product/9310113900069

We may need to put sane limits (1-2 digits only?); to avoid accidentally matching country labelling for some units like litres.

It may be safe to look for any matches > 100g / 100ml

**Comment by raphael0202** [comment]
We currently look at units such as "200 ml". "1 L" raised too many false positive to be used.
See https://github.com/openfoodfacts/robotoff/blob/e5534b488a988c23cc2c56fbf59d01384d7130c5/robotoff/insights/ocr/product_weight.py#L76
I'm sure the product weight detection logic can be improved though
